toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -91,7 +99,7 @@ func fetchPackages() (err error) {
 	}
 
 	hasUnresolvedNodes := hasUnresolvedNodes(dependencyGraph)
-	if hasUnresolvedNodes {
+	if *tryDownloadDeltaRPMs || hasUnresolvedNodes {
```

Comment:
>*tryDownloadDeltaRPMs || hasUnresolvedNodes [](http://example.com/codeflow?start=4&length=43)

I think we could now clean-up a little and limit what `fetchPackages` is doing to only fetching packages. What I have in mind:
- Move reading and writing the dependency graph out of this function.
- Move the `*tryDownloadDeltaRPMs || hasUnresolvedNodes` check outside this function as well and don't run it at all, in case the statement is false.
- Remove the two `*tryDownloadDeltaRPMs || hasUnresolvedNodes` checks inside `fetchPackages` since now we're sure that statement is true.

Reply:
I also grabbed the cloner setup code and put it into `setupCloner()`